### DISCUSSION TITLE
MTB 22.8.0 SQL Exception on tour upload / Unhandled event Loop

### DISCUSSION THREAD

**Discussion by stvzldn** [mytourbook/mytourbook#936] [General]
Sometimes MTB will start with a file upload, but not every time.  On first start today MTB did attempt to upload tours.  It immediately failed as detailed in the attached images when I selected "Save".
However after I clicked through all the failure messages and closed MTB, followed by restart of MTB, all seemed copacetic.  I did a Help->Database Consistency Check which reported no errors.
![Err MTB 1](https://user-images.githubusercontent.com/106929478/191568924-345fca0b-b6ed-43a0-9743-f91e7f7ab6f8.jpg)

![Err MTB 2](https://user-images.githubusercontent.com/106929478/191568930-3a76504d-b33c-45b0-ab11-a8d215a4d488.jpg)

![Err MTB 3](https://user-images.githubusercontent.com/106929478/191568951-4c19c270-da9d-402d-be81-c35e08bc2783.jpg)

![Err MTB 4](https://user-images.githubusercontent.com/106929478/191568956-00f07495-c91a-4445-9bb5-21dda8d4042b.jpg)

**Comment by wolfgang-ch** [comment]
Can you attach the log file(s) which can find in `C:\Users\YOURNAME\mytourbook\.metadata\*.log`

**Reply by stvzldn** [reply]
I do not find any .log but there are these .bak_* files and a unnamed text file.  In the attached .zip

[logfiles.zip](https://github.com/mytourbook/mytourbook/files/9625435/logfiles.zip)

**Reply by wolfgang-ch** [reply]
The log messages `org.eclipse.swt.SWTException: Invalid thread access` are causing subsequent errors, so first the invalid thread access errors must be fix.

**Comment by wolfgang-ch** [comment]
Created issue https://github.com/mytourbook/mytourbook/issues/938
